src/processes/parkingspaces/internal/create-note-of-interest.ts

```diff
@@ -291,24 +328,46 @@ export const createNoteOfInterestForInternalParkingSpace = async (
       }
     }
 
-    logger.error(
-      listingAdapterResult,
+    return endFailingProcess(
+      log,
+      CreateNoteOfInterestErrorCodes.InternalError,
+      500,
       'Create not of interest for internal parking space failed due to unknown error'
     )
-    return makeProcessError('internal-error', 500, {
-      message: 'failed due to unknown error',
-    })
   } catch (error: any) {
-    logger.error(
-      error,
-      'Create not of interest for internal parking space failed'
-    )
-    return makeProcessError('internal-error', 500, {
-      message: error.message,
-    })
+    const errorMessage =
+      error instanceof Error
+        ? 'Create not of interest for internal parking space failed: ' +
+          error.message
+        : 'Create not of interest for internal parking space failed: ' + error
+
+    return endFailingProcess(log, 'internal-error', 500, errorMessage)
   }
 }
 
+// Ends a process gracefully by debugging log, logging the error, sending the error to the dev team and return a process error with the error code and details
+const endFailingProcess = (
```

Comment:
It would be nice to move this function to a generic function in `src/processes/parkingspaces/utils.ts` since we have this code duplicated in the processes now

Reply:
Jag vet men jag vill ha olika titles i mailen beroende på vilken process som anropar och jag tyckte det blev så knöligt med ännu fler parametrar att skicka med varje gång